index.js

```diff
@@ -443,58 +459,93 @@ class HtmlWebpackInlineSVGPlugin {
+     * @param {Object{}} html
+     * @param {Object{}} inlineImage - parse5 document
+     * @param {Object{}} data - SVG data
+     * @param {Object{}} resolve - Consumer's Promise resolve hook
      * @returns {Promise}
      *
      */
-    processOutputHtml (html, inlineImage) {
+    optimizeSvg ({ html, inlineImage, data, resolve }) {
+        const configObj = Object.assign(svgoDefaultConfig, this.userConfig)
 
-        return new Promise((resolve, reject) => {
+        const config = {}
 
-            const svgSrc = this.getImagesSrc(inlineImage)
 
+        // pass all objects to the config.plugins array
 
-            // if the image isn't valid resolve
+        config.plugins = _.map(configObj, (value, key) => ({ [key]: value }));
 
-            if (!svgSrc) return resolve(html)
 
+        // create a new instance of SVGO
+        // passing it the merged config, to optimize the svg
 
-            // read in the svg
+        const svgo = new SVGO(config)
 
-            fs.readFile(path.resolve(this.outputPath, svgSrc), 'utf8', (err, data) => {
+        svgo.optimize(data)
+            .then((result) => {
+
+                const optimisedSVG = result.data
 
-                if (err) reject(err)
+                html = this.replaceImageWithSVG(html, inlineImage, optimisedSVG)
 
-                const configObj = Object.assign(svgoDefaultConfig, this.userConfig)
+                resolve(html)
 
-                const config = {}
+            })
+            .catch((err) => console.log(chalk.red(err)))
+    }
 
 
-                // pass all objects to the config.plugins array
+    /**
+     * append the inlineImages SVG data to the output HTML and remove the original img by
+     * loading the SVG data from the filesystem or from an URL
+     * @param {string} html
+     * @param {Object} inlineImage - parse5 document
+     * @returns {Promise}
+     *
+     */
+    processOutputHtml (html, inlineImage) {
 
-                config.plugins = _.map(configObj, (value, key) => ({ [key]: value }));
+        return new Promise((resolve, reject) => {
 
+            const svgSrc = this.getImagesSrc(inlineImage)
 
-                // create a new instance of SVGO
-                // passing it the merged config, to optimize the svg
 
-                const svgo = new SVGO(config)
+            // if the image isn't valid resolve
+            if (!svgSrc) return resolve(html)
 
-                svgo.optimize(data)
-                    .then((result) => {
+            // read in the svg
+            fs.readFile(path.resolve(this.outputPath, svgSrc), 'utf8', (err, data) => {
+                if (!err) {
+                    this.optimizeSvg({ html, inlineImage, data, resolve })
+                    return
+                }
 
-                        const optimisedSVG = result.data
+                // loading from the filesystem failed
+                if (!this.allowFromUrl) {
+                    reject(err)
```

Comment:
@AntonioRedondo If grabbing from the filesystem fails, then we reject and miserably fail. That's correct.